flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/snapshot/RocksIncrementalSnapshotStrategy.java

```diff
@@ -349,6 +350,13 @@ private long uploadSnapshotFiles(
                                 ? CheckpointedStateScope.EXCLUSIVE
                                 : CheckpointedStateScope.SHARED;
 
+                // Report the reuse of state handle to stream factory, which is essential for file
+                // merging mechanism.
+                checkpointStreamFactory.reusePreviousStateHandle(
```

Comment:
If the following uploading fails, the logic files will be deleted with delay?

And I think the `reusePreviousStateHandle` is a little bit indirect. In fact, the `lastUsedCheckpoint` of the newly generated and previous logical files are both updated here.

Reply:
Actually, only previous logical files are updated here, the `sstFiles` stands for the previous handle at this position.

Reply:
If the following uploading fails, can the logical files be deleted immediately? instead of deleting when receiving checkpoint abort notification.

Reply:
Nope. This could be optimized later. (This PR considers the file reusing only, not newly introduced files)